### PR TITLE
TicTacToe-GUI using Pygame 🐍

### LINKED ISSUE #768: Tic Tac Toe game with GUI using Pygame in Python
# Aim

- To make Tic Tac Toe using pygame
- 
# Details

- The game will be having a GUI with a single player and multiplayer feature.

@avinashkranjan I am a GSSoC'21 participant and I think this is an interesting python script. Kindly allow me to work on this issue and contribute to this project!

### PR DESCRIPTION
**Pull request by Lakhankumawat**
# Description
Added Tic Tac Toe game using GUI 
1. Making a GUI with pygame lib.
2. Building Game Logic .
3. Drawing squares and circles where ever player clicks.
4. Implementing 
5. Playing

# Details 
Libraries Used : pygame , numpy , sys

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/Lakhankumawat/Amazing-Python-Scripts/blob/TicTacToe-GUI/TicTacToe-GUI/README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Review by santushtisharma10 — CHANGES_REQUESTED** [review]
@Lakhankumawat display player 1 and player 2 turns while playing game. When the game ends display the results of game like which player has won or the result is tie. After the game ends add an option to restart or exit the game

**Comment by Lakhankumawat** [comment]
Okay i will work on requested changes.

**Comment by Lakhankumawat** [comment]
snip

Sir Requested Changes are made . Try to run it on your local machine code works fine. For further changes let me know .

**Comment by Lakhankumawat** [comment]
Please Ignore first commit consider second one which is just now .
First Commit had some issues
All requested changes are solved except one
> Tie is not detected even if you see i have commited somewhere to check a tie but it's not working rest everything works fine.

**Comment by Lakhankumawat** [comment]
No replies from @santushtisharma10 ma'am since 2 days @kaustubhgupta sir please have a look

File: TicTacToe-GUI/TicTacToe.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+# -*- coding: utf-8 -*-
+"""
+Tic Toe Using pygame , numpy and sys with Graphical User Interface
+
+"""
+
+import pygame, sys
+from pygame.locals import *
+import numpy as np
+#import WorkingModal as Wp
```
Comment (kaustubhgupta) [inline TicTacToe-GUI/TicTacToe.py:17]: remove this comment

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Bug:

When the game is a draw nothing happens:
image

File: TicTacToe-GUI/TicTacToe.py (hunk 1 above)
Comment (Lakhankumawat) [reply TicTacToe-GUI/TicTacToe.py:17]: > Please Ignore first commit consider second one which is just now .
> First Commit had some issues
> All requested changes are solved except one
> 
> > Tie is not detected even if you see i have commited somewhere to check a tie but it's not working rest everything works fine.

Yeah i mentioned it here i will try removing this bug.

**Comment by Lakhankumawat** [comment]
ss

Looks Fine!!

**Comment by santushtisharma10** [comment]
@Lakhankumawat apologies. Was busy with other some stuff. Will review it shortly

**Comment by Lakhankumawat** [comment]
> @Lakhankumawat apologies. Was busy with other some stuff. Will review it shortly

Okay no problems you guys already seemingly work hard to merge everyone's PR understanding them and reviewing it's not simple . Take your time

**Review by kaustubhgupta — APPROVED** [review]
Perfect 💯

**Review by santushtisharma10 — APPROVED** [review]
Nice @Lakhankumawat 💯💯

File: TicTacToe-GUI/TicTacToe.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+                    Won= font.render("Player"+str(player)+" Won ", True ,blue,green)
+                    screen.blit(Won, winRect)
+                    screen.blit(text, textRect) 
+                    screen.blit(leave,leaveRect)
+                player=player%2 +1
+                if not game_over and not board_full:
+                    Won= font.render("Player"+str(player)+" Turn ", True ,blue,green)
+                    screen.blit(Won, winRect)
+                draw_figures()
+
```
Comment (antrikshmisri) [inline TicTacToe-GUI/TicTacToe.py:213]: Please remove the extra blank lines

**Review by antrikshmisri — CHANGES_REQUESTED** [review]

File: TicTacToe-GUI/TicTacToe.py (hunk 2 above)
Comment (kaustubhgupta) [reply TicTacToe-GUI/TicTacToe.py:213]: @Lakhankumawat Please make these changes so that we can move your PR to PA

**Review by antrikshmisri — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM..🚩

**Pull request merged**